File: frontend/src/types/report.ts
// ---- Report Summary (list item) ----
export interface ReportSummary {
  id: string;
  type: 'daily' | 'weekly';
  date_range_start: string;
  date_range_end: string;
  file_path: string;
  paper_count: number;
  repo_count: number;
  delivery_status: 'pending' | 'delivered' | 'failed';
  generated_at: string;
}

// ---- Report Detail (with content) ----
export interface ReportDetail extends ReportSummary {
  content: string;
}

// ---- Report Query Params ----
export interface ReportQueryParams {
  page?: number;
  page_size?: number;
  type?: 'daily' | 'weekly';
  sort?: string;
}

File: backend/app/schemas/report.py
"""Report Pydantic schemas."""

from datetime import date, datetime

from pydantic import BaseModel


class ReportSummary(BaseModel):
    id: str
    type: str
    date_range_start: date
    date_range_end: date
    file_path: str
    paper_count: int
    repo_count: int
    delivery_status: str
    generated_at: datetime

    model_config = {"from_attributes": True}


class ReportDetail(ReportSummary):
    content: str
